chez-libs/unison/primops.ss

```diff
@@ -145,11 +167,8 @@
   (define (unison-POp-FTOT f) (number->istring f))
   (define (unison-POp-IDXB n bs) (bytevector-u8-ref bs n))
   (define (unison-POp-IDXS n l)
-    (call/1cc
-      (lambda (k)
-        (with-exception-handler
-          (lambda (e) (list 0))
-          (lambda () (list-ref l n))))))
+    (guard (x [else (list 0)])
+      (list 1 (list-ref l n))))
```

Comment:
`guard` seemed like a simpler api to catch exceptions, and the previous version wasn't working for me 🙃

Reply:
Yeah, somehow I missed `guard` when I was reading about exceptions (I guess because examples used `call/cc` instead for some reason). It's the portable solution, and exactly what we wanted anyway.